### PR TITLE
feat: added support for bar chart stacking 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

- added stacking options for bar charts 

#### Related Issues / PR's

fixes https://github.com/SigNoz/signoz/issues/4959

#### Screenshots

https://github.com/SigNoz/signoz/assets/54737045/5baeafa4-c781-4997-9474-043ad860058f

### CODE REVIEW

**Review by SagarRajput-7 — DISMISSED** [review]
LGTM

File: frontend/src/lib/uPlotLib/getUplotChartOptions.ts [hunk 1]
```diff
@@ -42,6 +44,68 @@ export interface GetUPlotChartOptions {
...
+		[key: string]: boolean;
+	};
+	setHiddenGraph?: Dispatch<
+		SetStateAction<{
+			[key: string]: boolean;
+		}>
+	>;
+}
+
+function getStackedSeries(apiResponse: QueryData[]): QueryData[] {
```
Comment (YounixM) [inline frontend/src/lib/uPlotLib/getUplotChartOptions.ts:58]: Add details around the implementation

File: frontend/src/lib/uPlotLib/getUplotChartOptions.ts [hunk 2]
```diff
@@ -42,6 +44,68 @@ export interface GetUPlotChartOptions {
...
+				parseFloat(values[j][1]) + parseFloat(series[i + 1].values[j][1]),
+			);
+		}
+
+		series[i].values = values;
+	}
+
+	return series;
+}
+function getStackedSeriesQueryFormat(apiResponse: QueryData[]): QueryData[] {
```
Comment (YounixM) [inline frontend/src/lib/uPlotLib/getUplotChartOptions.ts:74]: Add details around the implementation

File: frontend/src/lib/uPlotLib/plugins/tooltipPlugin.ts [hunk 3]
```diff
@@ -67,7 +68,13 @@ const generateTooltipContent = (
 					unit = '',
 				} = seriesList[index - 1] || {};
 
-				const value = data[index][idx];
+				// eslint-disable-next-line no-nested-ternary
+				const value = stackBarChart
+					? index + 1 < data.length
```
Comment (YounixM) [inline frontend/src/lib/uPlotLib/plugins/tooltipPlugin.ts:73]: Can right this cleaner.

File: frontend/src/lib/uPlotLib/getUplotChartOptions.ts (hunk 1 above)
Comment (vikrantgupta25) [reply frontend/src/lib/uPlotLib/getUplotChartOptions.ts:64]: done

File: frontend/src/lib/uPlotLib/getUplotChartOptions.ts (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/lib/uPlotLib/getUplotChartOptions.ts:84]: done

File: frontend/src/lib/uPlotLib/plugins/tooltipPlugin.ts (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/lib/uPlotLib/plugins/tooltipPlugin.ts]: moved this to a util function and added it as if else rather than ternary

**Comment by YounixM** [comment]
Screenshot 2024-06-10 at 2 00 35 PM

The highest bar doesnt have filled color when other bars are deselected.

**Comment by YounixM** [comment]
Dashboard Edit View (generic issue) - There is shift in width when we add a 2nd query

**Comment by vikrantgupta25** [comment]
> Dashboard Edit View (generic issue) - There is shift in width when we add a 2nd query

will pick this up separately 👍

**Comment by vikrantgupta25** [comment]
> Screenshot 2024-06-10 at 2 00 35 PM
> The highest bar doesnt have filled color when other bars are deselected.

- fixed the issue, we will not have the graph manager in the bar charts, the uplot hide and show for bar charts stacked is not built on multiple series. we show and hide only one legend at a time. built in sync with other products .

**Comment by vikrantgupta25** [comment]
image

added the info text

**Comment by vikrantgupta25** [comment]
> Dashboard Edit View (generic issue) - There is shift in width when we add a 2nd query

Fixed here - https://github.com/SigNoz/signoz/pull/5182

**Review by YounixM — APPROVED** [review]

**Pull request merged**